core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -127,6 +150,29 @@ class RecordInsightsLOCO[T <: Model[T]]
     left.zipAll(right, 0.0, 0.0).map { case (l, r) => l + r }
   }
 
+  /**
+   * Optionally convert columnMetadata's descriptorValue like
+   * "y_DayOfWeek", "x_DayOfWeek" to TimePeriod enum - DayOfWeek.
+   * @return Option[TimePeriod]
+   */
+  private def convertToTimePeriod(descriptorValue: String): Option[TimePeriod] =
+    descriptorValue.split("_").lastOption.flatMap(TimePeriod.withNameInsensitiveOption)
+
+  private def isMapFeature(featureType: String ): Boolean = {
+    val featureTypeTag = FeatureType.featureTypeTag(featureType)
+    featureTypeTag.tpe <:< weakTypeOf[OPMap[_]]
```

Comment:
this check is very very very slow to perform at runtime. let's just check if `history.grouping` is present or not. ie. 

```scala
private def getRawFeatureName(history: OpVectorColumnHistory): Option[String] = history.grouping match {
    case Some(grouping) => history.parentFeatureOrigins.headOption.map(_ + "_" + grouping)
    case None => history.parentFeatureOrigins.headOption
}
```